### PR TITLE
enhanced ui for cheatsheets and footer fix and error 404 and intervie…

### PR DESCRIPTION
**Pull request by abhishekpanthee**
Fix the UI to make visually appealing for dark mode
Working clickable redirects or navigation from footer to the descriptive section
One in all footer for every section that include all the desired sections to navigate
additional fix 
- fixed Error 404 page 
- FIxed  machinecoding subpage ui 

BEFORE -- 

b9
b8
b7
b6
b5
b4
b3
b2
b1

AFTER --- 
Screenshot from 2024-10-04 23-47-48
Screenshot from 2024-10-04 23-47-37
Screenshot from 2024-10-04 23-47-33
Screenshot from 2024-10-04 23-47-26
Screenshot from 2024-10-04 23-47-22
Screenshot from 2024-10-04 23-47-15
Screenshot from 2024-10-04 23-47-11
Screenshot from 2024-10-04 23-47-06
Screenshot from 2024-10-04 23-46-57
Screenshot from 2024-10-04 23-47-48

### CODE REVIEW

**Comment by abhishekpanthee** [comment]
also could you please add hacktoberfest hacktoberfest-accepted label  to the repo it would also be helpful 
Screenshot from 2024-10-05 17-58-19
@arjunatapadkar

File: src/pages/ErrorPage.jsx
Comment (arjunatapadkar) [inline src/pages/ErrorPage.jsx:1]: Only this page is acceptable

**Comment by arjunatapadkar** [comment]
@abhishekpanthee are you resolving or should I close the PR

**Comment by abhishekpanthee** [comment]
oh sorry i couldn't get to you so should i only modify error page or resolve merge conflicts i couldn't get what you are saying 
should I push only error page 
@arjunatapadkar

**Comment by arjunatapadkar** [comment]
@abhishekpanthee  Yeah only push error page and then contribute to other whatever issues you want to one by one

**Pull request closed without merging**